File: app/src/main/java/the/mic/swag/ui/fragment/alert/AlertFragment.java
package the.mic.swag.ui.fragment.alert;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.fragment.app.Fragment;

import the.mic.swag.R;

public class AlertFragment extends Fragment {

    public AlertFragment() {

    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {

        return inflater.inflate(R.layout.fragment_alert, container, false);
    }
}

File: app/src/main/java/the/mic/swag/ui/fragment/alert/AlertFragment.kt
package the.mic.swag.ui.fragment.alert

import android.os.Bundle
import android.view.View
import androidx.fragment.app.FragmentManager
import androidx.lifecycle.ViewModelProviders
import com.android.dutchman.util.DataBindingFragment
import dsm.android.v3.presentation.di.scope.ActivityScope
import the.mic.swag.R
import the.mic.swag.databinding.FragmentAlertBinding
import the.mic.swag.domain.repository.alert.AlertRepository
import the.mic.swag.presentation.viewmodel.alert.AlertViewModel
import the.mic.swag.presentation.viewmodel.alert.AlertViewModelFactory

@ActivityScope
class AlertFragment : DataBindingFragment<FragmentAlertBinding>(), AlertRepository {

    override val layoutId: Int
        get() = R.layout.fragment_alert

    private val factory = AlertViewModelFactory(this)

    private val fm: FragmentManager? by lazy { fragmentManager }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        val viewModel = ViewModelProviders.of(activity!!, factory).get(AlertViewModel::class.java)

        binding.vm = viewModel

    }


}